### ISSUE TITLE
Working code?

### ISSUE THREAD

**Issue by XMaarten** [michaelusner/esp32_pentair_controller#1]
I would like to do the same thing at my home. Is your project already in a working state? 

Thanks, Maarten

**Comment by michaelusner** [comment]
@XMaarten ,

Yes, the code is working and is running on my ESP32 sitting in front of me.  It's integrated with HomeAssistant and I'm able to control via my phone.I am seeing an issue where after a few days, it will stop controlling equipment.  When I connect to the ESP32's serial port, it will start working again so it's hard to debug.
I have a few changes I wanted to make but it it is in a "mostly" working state today.

I'll add a picture of the controller in a moment.

**Comment by michaelusner** [comment]
I'm working on an ESPHome version of this project that should allow more flexibility and better integration in Home Assistant. It's still a WIP but I'll update the issues here when it's ready.

**Comment by XMaarten** [comment]
Are you making any progress with your ESPHome version? Could you maybe publish the work you done so far in a development branch so others, like me, can help you with development?

I found another project that seems to be useful. I think it is for a different protocol, but is for integrating a pump with ESPHome as well. It might be useful to work together. See [CentryVSPump](https://github.com/gazoodle/CenturyVSPump)

**Comment by michaelusner** [comment]
Hi Maarten,

I had to replace my main Easytouch board after lightning destroyed it.  I
also added a ScreenLogic2 controller so the urgency on this project isn't
so great anymore.  I also found that the com portion of my pump is not
working either and will have to replace the pump drive.  I'm using a
manually programmed schedule for the pump now but I think I'm at the point
where I can resume my work on this project.

My plan was to create a custom ESPHome component where the pool device
RS485 addresses can be defined in the yaml file.  I have a basic component
that reads air and water temperature and displays in Home Assistant but am
having a bit of trouble adding a custom switch to the component for the
devices.
I have to say that I'm very discouraged at the lack of ESPHome
documentation and examples for a use-case like this.

Anyway, I'll continue here and once I solve the custom switch problem, I
should be able to make good progress.

Thanks,
Mike

**Comment by XMaarten** [comment]
Did you have a look at the project I mentioned? It is the same thing, an ESPHome component for a RS485 controlled pump. I think it is using a different protocol, but the functionality seems to be the same.

**Comment by michaelusner** [comment]
I'll have a closer look.  It seems to be a bit different than what I have
but might have some useful info.
I elected not to use the modbus component in my case and instead derived
from the UARTDevice class.

Thank you for the link BTW.

-Mike

**Comment by michaelusner** [comment]
I think things are finally working again for me.  The lighting also destroyed my ESP32 dev board and MAX485 so I had to replace them.
I have temperature coming through fine and am working on switches for devices and components now:
![image](https://github.com/michaelusner/esp32-pentair-controller/assets/6410647/3173897d-5316-4cfb-81ed-c87aff7ed78a)

**Comment by XMaarten** [comment]
Great! Can you already share your code please? I am very curious if can get some results as well.

**Comment by michaelusner** [comment]
Sure, let me clean things up a bit and I'll put up a draft.
Note that the switch doesn't work yet.

**Comment by michaelusner** [comment]
@XMaarten https://github.com/michaelusner/esphome-components

**Comment by NumanGulzar** [comment]
I will also get a sneeze to reduce??
